uninstall.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/** @var rex_addon $this */
+
+// update config
+// remove additional config from base config
+$config = array_diff_recursive(
+    rex_file::getConfig($this->getPath('package.yml')),
+    rex_file::getConfig($this->getPath('package.setup.yml'))
+);
+
+rex_file::putConfig($this->getPath('package.yml'), $config);
+
+
+// Computes the difference of two arrays recursively
+// https://gist.github.com/t3chnik/6b3b14d3859d810c02f4
+function array_diff_recursive($aArray1, $aArray2)

```

Comment:
Die Funktion mit diesem sehr allgemeinen Namen einfach so im globalen Namespace zu deklarieren, finde ich nicht so gut.
Kann durchaus passieren, dass die Funktion schon woanders definiert ist. Zum Beispiel wenn die in mehreren Addons in der uninstall ist, und es mal eine Situation gäbe, wo mehrere Addons gleichzeitig deinstalliert werden (Ok, gibt es aktuell glaube ich nicht), dann würde es krachen.

Man könnte ein `function_exists` drum machen. Dann muss man sich aber auch drauf verlassen, dass falls sie schon definiert ist, sie genauso arbeitet.
Alternativ könnte man einen spezifischeren Funktionsnamen nehmen. Oder mit einer anonymen Funktion, wobei das bei rekursiven Funktionen etwas tricky ist:
```php
$arrayDiffRecursive = function (aArray1, aArray2) use (&$arrayDiffRecursive) {
    // ...
    $arrayDiffRecursive(...);
    // ...
}

Reply:
oder einen namespace verwenden.

Reply:
Zum Hintergrund: Die `install.php` und `uninstall.php` werden in allen Demos auf gleiche Art verwendet und sollten einfach hin- und herkopierbar sein, ohne AddOn-spezifische Anpassungen vornehmen zu müssen (wie Namespaces). Wenn ihr eine gute Lösung habt, bringt sie gerne ein. Ich bin nicht so firm im Thema, um das sicher zu lösen.

Reply:
wenn du sie 1:1 kopieren willst, würde ich die funktion in eine lokale variable schreiben - wie gregor oben vorschlägt.